core/service/team_test.go

```diff
@@ -0,0 +1,804 @@
+
+	teamyPrometheus := metricstest.NewNoopMetrics()
+	cloudClientCfg := rpc.ConnectionConfig{
+		Host:          testkit.GRPCServerHost,
+		Port:          testkit.GRPCServerPort,
+		ShouldEncrypt: false,
+		GetAccessToken: func() string {
+			return "accessToken"
+		},
+		RequestTimeout: 10 * time.Second,
+	}
+	cloudClientRegistry, err := cloudAPI.NewClientRegistry(
+		dataCollector,
+		virtualNetwork,
+		teamyPrometheus,
+		cloudClientCfg,
+		func() retry.Retry {
+			exponentialBackOff := backoff.NewExponentialBuilder().Build()
+			return retry.NewMaxCount(runtime.NewBuiltInRuntime(), exponentialBackOff, 3)
+		})
+	if !assert.Nil(t, err) {
+		return Team{}, false
+	}
+
+	authorizer := NewAuthorizer(dataCollector, cloudClientRegistry)
+
+	transactionFactory := transaction.NewFactory(nil)
+
+	teamyBackendDB := dbtest.NewInMemoryDB()
+	teamyBackendDB.CreateTable(daotestv2.TeamTableName)
+	teamyBackendDB.CreateTable(daotestv2.TeamMemberTableName)
+	teamyBackendDB.CreateTable(daotestv2.TeamFileUploadSessionTableName)
+	teamyBackendDB.CreateTable(daotestv2.SprintTableName)
+
+	teamMemberDao := daotest.NewTeamMember(teamyBackendDB)
+	teamMemberDaoV2 := daotestv2.NewTeamMember(teamyBackendDB, transactionFactory)
+	stateSyncer := realtime.NewStateSyncer(dataCollector, teamMemberDao, teamMemberDaoV2)
+
+	taskDaoV2 := daotestv2.NewTask(teamyBackendDB)
+	sprintDao := daotest.NewSprint(teamyBackendDB)
+	sprintDaoV2 := daotestv2.NewSprint(teamyBackendDB, transactionFactory)
+	sprintParticipantDao := daotest.NewSprintParticipant(teamyBackendDB)
+	sprintParticipantDaoV2 := daotestv2.NewSprintParticipant(teamyBackendDB)
+	teamDao := daotest.NewTeam(teamyBackendDB)
+	teamDaoV2 := daotestv2.NewTeam(teamyBackendDB, transactionFactory)
+	teamFileUploadSessionDaoV2 := daotestv2.NewTeamFileUploadSession(teamyBackendDB)
+
+	return NewTeam(
+		dataCollector,
+		cloudTestKitConfig.WebAPIBaseURL,
+		cloudClientRegistry,
+		authorizer,
+		stateSyncer,
+		transactionFactory,
+		taskDaoV2,
+		sprintDao,
+		sprintDaoV2,
+		sprintParticipantDao,
+		sprintParticipantDaoV2,
+		teamDao,
+		teamDaoV2,
+		teamMemberDao,
+		teamMemberDaoV2,
+		teamFileUploadSessionDaoV2,
+	), true
+}
+
+func TestTeamService_FindTeamByID(t *testing.T) {
+	teamService, ok := prepareTeamService(t)
+	if !ok {
+		return
+	}
+
+	var requesterUserID uint64 = 20
+	var teamID uint64 = 12
+	ct := context.Background()
+	ct = ctx.NewContextWithUserID(ct, requesterUserID)
+
+	tx, err := teamService.transactionFactory.BeginTx(ct, nil)
+	if !assert.Nil(t, err) {
+		return
+	}
+	defer tx.Rollback()
+
+	var iconURL = "https://test"
+	now := time.Now().UTC()
+	team := entity.Team{
+		ID:            teamID,
+		Name:          "TeamName",
+		IconURL:       &iconURL,
+		CreatorUserID: requesterUserID,
+		OwnerUserID:   requesterUserID,
+		CreatedAt:     now,
+		UpdatedAt:     &now,
+	}
+
+	// insert team into table
+	if !assert.Nil(t, teamService.teamDaoV2.CreateTeam(ct, tx, team)) {
```

Comment:
```suggestion
	if !assert.Nil(t, refs.teamDaoV2.CreateTeam(ct, tx, team)) {
```

Let's return `Ref` struct in `prepareTeamService` and use `teamDaoV2` from `refs` instead to avoid breaking encapsulation.